airflow/bin/cli.py

```diff
@@ -607,6 +636,10 @@ class CLIFactory(object):
         'execution_date': Arg(
             ("execution_date",), help="The execution date of the DAG",
             type=parsedate),
+        'execution_date_enhanced': Arg(
```

Comment:
I went that way first. At the point where the type converter (e.g. int, parsedate, etc...) is used from the argparse module, I don't have access to the dag. Hence, I need to execute these conversions later in code execution